tests/load/test_read_interfaces.py

```diff
@@ -677,6 +680,115 @@ def test_column_selection(populated_pipeline: Pipeline) -> None:
         arrow_table = table_relationship.select("unknown_column").head().arrow()
 
 
+@pytest.mark.no_load
+@pytest.mark.essential
+@pytest.mark.parametrize(
+    "populated_pipeline",
+    configs,
+    indirect=True,
+    ids=lambda x: x.name,
+)
+def test_select(populated_pipeline: Pipeline) -> None:
+    table_relationship = populated_pipeline.dataset(dataset_type="default").items
+
+    columns = ["_dlt_load_id", "other_decimal"]
+    data_frame = table_relationship.select(*columns).head().df()
+    assert list(data_frame.columns.values) == columns
+
+    data_frame = table_relationship.select(*columns).select(*columns).head().df()
+    assert list(data_frame.columns.values) == columns
+
+    data_frame = table_relationship.select(*columns).select(*["other_decimal"]).head().df()
+    assert list(data_frame.columns.values) == ["other_decimal"]
+
+    with pytest.raises(LineageFailedException):
+        data_frame = table_relationship.select(*columns).select(*["decimal"]).head().df()
+
+
+@pytest.mark.no_load
+@pytest.mark.essential
+@pytest.mark.parametrize(
+    "populated_pipeline",
+    configs,
+    indirect=True,
+    ids=lambda x: x.name,
+)
+def test_order_by(populated_pipeline: Pipeline) -> None:
+    total_records = _total_records(populated_pipeline.destination.destination_type)
+    table_relationship = populated_pipeline.dataset(dataset_type="default").items
+
+    asc_ids = [row[0] for row in table_relationship.order_by("id", "asc").limit(20).fetchall()]
+    assert asc_ids == list(range(20))
+
+    desc_ids = [row[0] for row in table_relationship.order_by("id", "desc").limit(20).fetchall()]
+    assert desc_ids == list(range(total_records - 1, total_records - 21, -1))
+
+    chained = [row[0] for row in table_relationship.order_by("id").limit(5).select("id").fetchall()]
+    assert chained == list(range(5))
+
+    @dlt.resource
+    def chained_order():
```

Comment:
the pipelines dataset are not wiped between tests to make them faster, so adding a new resource here has broken other tests.